web/src/components/Upstream/components/Type.tsx

```diff
@@ -52,7 +52,7 @@ const CHash: React.FC<Pick<Props, 'readonly'>> = ({ readonly }) => {
       </Form.Item>
       <Form.Item
         name="key"
-        rules={[{ required: true }]}
+        rules={[{ required: form.getFieldValue('hash_on') !== 'consumer' }]}
         label={formatMessage({ id: 'component.upstream.fields.key' })}
```

Comment:
it would be better to add `disabled` attr to `key` field when user selected consumer, or hide this field is also a good choice

for when hash_on selected consumer, key will auto match current request's consumer automatically  no matter what user entered